Fix encoding of LatinRules.xdy (now UTF-8)

alcrene opened the pull request:
### Feature or Bugfix
- Bugfix

### Purpose
At the moment, sphinx/texinputs/LatinRules.xdy is encoded as
(as reported by `file LatinRules.xdy`):

  > Non-ISO extended-ASCII text, with LF, NEL line terminators

This has at least two consequences:
1. More basic text readers display the file incorrectly. This includes
   the file's own [raw view](https://raw.githubusercontent.com/sphinx-doc/sphinx/31eba1a76dd485dc633cae48227b46879eda5df4/sphinx/texinputs/LatinRules.xdy) on GitHub.
2. Certain pipelines may automatically convert this to UTF-8 (ostensibly
   a good thing), which causes headaches with diff and version control
   (500 spurious line changes every time the document is rebuilt).

This commit simply re-encodes the file to UTF-8. (Compare [raw view after change](https://raw.githubusercontent.com/alcrene/sphinx/fix-LatinRules-encoding/sphinx/texinputs/LatinRules.xdy).)

jfbu reviewed (commented):
The proposed change will break the purpose of the file. See detailed comment above.

alcrene committed:
Explain non-standard encoding in LatinRules.xdy

In order to fulfill its function, LatinRules.xdy must use single,
non-standard byte characters (neither ASCII, nor multi-byte UTF-8).
To someone encountering the file without knowing its purpose (e.g.
due a post-processing raising a warning for the unrecognized encoding)
this is likely surprising, and may seem like a holdover from a time where
Unicode wasn't as universally supported.

The added comment should make clear that the file must stay as it is,
and in particular that it must not be "standardized" to UTF-8.

jfbu commented on sphinx/texinputs/LatinRules.xdy:
```diff
@@ -1,6 +1,10 @@
 ;; filename: LatinRules.xdy
 ;;
+;; Please note that this data file deliberately uses strings
+;; with single non-ascii bytes. This is intentional and
```
```suggestion
;; with single non-ascii bytes.  This is intentional and
```

jfbu reviewed (changes requested):
Thanks for contribution! Can you add a space as a sentence is ended by two spaces in the existing comments?

(sorry for nitpicking ;-) btw I notice that some sentences  n pre-existing comments sometimes lack a full stop)

alcrene committed:
Fix punctuation in comments

jfbu reviewed (approved):
LGTM, thanks!
